app/views/submissions.scala.html

```diff
@@ -46,6 +47,14 @@
     }
 }
 
+@warningText = @{
+    if (submissionsPagesElements.actionSubmissions.currentPageElements.exists(_._2.exists(_.status == SubmissionStatus.QUERY_NOTIFICATION_MESSAGE))) {
```

Comment:
Just to simplify a little bit, maybe something like
```
val element = submissionsPagesElements.actionSubmissions.currentPageElements
if (element.exists(_._2.exists(_.status == QUERY_NOTIFICATION_MESSAGE))) {
```
importing `QUERY_NOTIFICATION_MESSAGE` ?

Reply:
Yes, it is a bit of hard line to parse. Perhaps this test should be done in the controller and just pass a `isQueryNotificationPresent` boolean to the view.
Could expand this logic in the view to something more readable like:
` val isQueryNotificationPresent = submissionsPagesElements.actionSubmissions.currentPageElements
      .exists{ submissionNotificationsPair =>
        submissionNotificationsPair._2.exists( notification => notification.status == SubmissionStatus.QUERY_NOTIFICATION_MESSAGE)
      }`